leanplum_data_export/export.py

```diff
@@ -245,19 +305,89 @@ def get_table_name(self, table_prefix, leanplum_name, version, date=None, datase
 
         return name
 
-    def add_slash_if_not_present(self, val):
-        if not val.endswith("/"):
-            val = f"{val}/"
-        return val
-
-    def parse_schema(self, schema_file_path):
-        with open(schema_file_path) as schema_file:
-            fields = json.load(schema_file)
-        return [
-            bigquery.SchemaField(
-                field["name"],
-                field_type=field.get("type", "STRING"),
-                mode=field.get("mode", "NULLABLE"),
-            )
-            for field in fields
-        ]
+    def parse_schema(self, data_type):
+        try:
+            with open(os.path.join(self.SCHEMA_DIR, f"{data_type}.schema.json"), "r") as schema_file:
+                return [field for field in json.load(schema_file)]
+        except FileNotFoundError:
+            raise ValueError(f"Unrecognized table name encountered: {data_type}")
+
+    @staticmethod
+    def get_gcs_prefix(prefix, version, date, data_type=None):
+        if data_type is None:
+            return os.path.join(prefix, f"v{version}", date, "")
+        else:
+            return os.path.join(prefix, f"v{version}", date, data_type, "")
+
+    @staticmethod
+    def extract_user_attributes(session_data):
```

Comment:
Nit: I would consider moving the functions that act on single JSON blobs to a separate file than the functions that act on entire JSON files.

Reply:
I agree